src/hotspot/share/runtime/safepoint.cpp

```diff
@@ -705,45 +705,34 @@ void SafepointSynchronize::block(JavaThread *thread) {
   }
 
   JavaThreadState state = thread->thread_state();
+  assert(SafepointSynchronize::is_a_block_safe_state(state), "Illegal threadstate encountered: %d", state);
```

Comment:
Nit: you shouldn't need to say SafepointSynchronize:: here.

Reply:
Agreed. Will fix it.